docs/posts/ibisml/index.qmd

```diff
@@ -0,0 +1,897 @@
+    * A neural network with PyTorch and PyTorch Lightning
+
+The aim of this competition is to predict which clients are more likely to default on their
+loans by using both internal and external data sources.
+
+To get started with Ibis and IbisML, please refer to the websites:
+
+* [Ibis](https://ibis-project.org/): An open-source dataframe library that works with any data system.
+* [IbisML](https://ibis-project.github.io/ibis-ml/): A library for building scalable ML pipelines.
+
+
+## Prerequisites
+To run this example, you'll need to download the data from Kaggle website with a Kaggle user account and install Ibis, IbisML, and the necessary modeling library.
+
+### Download data
+You need a Kaggle account to download the data. If you do not have one,
+feel free to register one.
+
+1. Option 1: Manual Download
+     * Log into your Kaggle account and download all data from this
+     [link](https://www.kaggle.com/competitions/home-credit-credit-risk-model-stability/data),
+     unzip the files, and save them to your local disk.
+2. Option 2: Kaggle API
+    * Go to your Kaggle account settings: [Kaggle Account Settings](https://www.kaggle.com/account).
+    * Under the "API" section, click on "Create New API Token". This will download the `kaggle.json`
+    file to your computer.
+    * Place the `kaggle.json` file in the correct directory, normally it is under your home directory
+    `~/.kaggle`:
+
+        ```bash
+        mkdir ~/.kaggle
+        mv ~/Downloads/kaggle.json ~/.kaggle
+        ```
+    * Install Kaggle CLI and download the data:
+
+        ```bash
+        pip install kaggle
+        kaggle competitions download -c home-credit-credit-risk-model-stability
+        unzip home-credit-credit-risk-model-stability.zip
+        ```
+
+### Install libraries
+To use Ibis and IbisML with the DuckDB backend, you'll need to install the necessary packages:
+```{.bash}
+pip install 'ibis-framework[duckdb]' ibis-ml
+```
+
+When building models, ensure you have the required libraries installed:
+```{.bash}
+# PyTorch
+pip install torch torchvision pytorch-lightning
+
+# XGBoost and sklearn
+pip install sklearn xgboost
+```
+## Data loading and processing
+We'll use Ibis to read the parquet files (with the same data also available in CSV format)
+ and perform the necessary processing for the next step.
+
+::: {.callout-tip}
+You can perform the same process with a different backend, such as
+Polars, BigQuery, Spark, or other Ibis-supported backends. Move the datasets to the
+desired database and set the Ibis backend using the reference: [Work with multiple
+backends](https://ibis-project.org/how-to/input-output/multiple-backends).
+:::
+
+### Directory structure and tables
+Since there are many data files, let's start by examining the directory structure and
+tables within the train directory:
+
+```bash
+# Change this to your directory
+tree -L 2 /Users/claypot/Downloads/home-credit-credit-risk-model-stability/parquet_files/train
```

Comment:
Nit: Does it make sense to swap `claypot` references to `jiting` or something? Not really important in the grand scheme of things, but seems like an easy change. :)